Add warning for network policy enforcement (#7967)
istio/istio.io: content/en/docs/tasks/traffic-management/egress/egress-gateway/index.md

    EOF
    {{< /text >}}

> {{< warning >}}

Reviewer: The warning needs to be indented 4 spaces, are it will mess up the numbering of the following steps.

[Network policies](https://kubernetes.io/docs/concepts/services-networking/network-policies/)
are implemented by the network plugin in your Kubernetes cluster.

> Depending on your test cluster, the traffic may not be blocked as described in
> this step.

Reviewer: ```suggestion
Depending on your test cluster, the traffic may not be blocked in the following step.
```
